src/textformatter.ts

```diff
@@ -1,346 +0,0 @@
-  }
-
-  /** Create a hash with the current font data, so we can cache computed widths. */
-  updateCacheKey(): void {
-    const family = this.family.replace(/\s+/g, '_');
-    const size = this.size;
-    const weight = this.bold ? FontWeight.BOLD : FontWeight.NORMAL;
-    const style = this.italic ? FontStyle.ITALIC : FontStyle.NORMAL;
-    // Use the same key format as SVGContext.
-    this.cacheKey = `${family}%${size}%${weight}%${style}`;
-  }
-
-  /**
-   * The glyphs table is indexed by the character (e.g., 'C', '@').
-   * See: robotoslab_glyphs.ts & petalumascript_glyphs.ts.
-   */
-  getGlyphMetrics(character: string): FontGlyph {
-    if (this.glyphs[character]) {
-      return this.glyphs[character];
-    } else {
-      return this.glyphs[this.maxSizeGlyph];
-    }
-  }
-
-  get maxHeight(): number {
-    const metrics = this.getGlyphMetrics(this.maxSizeGlyph);
-    return (metrics.ha / this.resolution) * this.fontSizeInPixels;
-  }
-
-  /**
-   * Retrieve the character's advanceWidth as a fraction of an `em` unit.
-   * For the space character ' ' as defined in the:
-   *   petalumascript_glyphs.ts: 250 advanceWidth in the 1000 unitsPerEm font returns 0.25.
-   *   robotoslab_glyphs.ts:     509 advanceWidth in the 2048 unitsPerEm font returns 0.2485.
-   */
-  getWidthForCharacterInEm(c: string): number {
-    const metrics = this.getGlyphMetrics(c);
-    if (!metrics) {
-      // An arbitrary number, close to the `em` width of the '#' and '5' characters in PetalumaScript.
-      return 0.65;
-    } else {
-      const advanceWidth = metrics.advanceWidth ?? 0;
-      return advanceWidth / this.resolution;
-    }
-  }
-  /**
-   * Retrieve the character's y bounds (ymin, ymax) and height.
-   */
-  getYForCharacterInPx(c: string): yExtent {
-    const metrics = this.getGlyphMetrics(c);
-    const rv = { yMin: 0, yMax: this.maxHeight, height: this.maxHeight };
-    if (!metrics) {
-      return rv;
-    } else {
-      if (typeof metrics.yMin === 'number') {
-        rv.yMin = (metrics.yMin / this.resolution) * this.fontSizeInPixels;
-      }
-      if (typeof metrics.yMax === 'number') {
-        rv.yMax = (metrics.yMax / this.resolution) * this.fontSizeInPixels;
-      }
-      rv.height = rv.yMax - rv.yMin;
-      return rv;
-    }
-  }
-  getYForStringInPx(str: string): yExtent {
-    const entry = this.localHeightCache;
-    const extent = { yMin: 0, yMax: this.maxHeight, height: this.maxHeight };
-    const cache = entry[str];
-    if (cache !== undefined) {
-      return cache;
-    }
-    for (let i = 0; i < str.length; ++i) {
-      const curY = this.getYForCharacterInPx(str[i]);
-      extent.yMin = Math.min(extent.yMin, curY.yMin);
-      extent.yMax = Math.max(extent.yMax, curY.yMax);
-      extent.height = extent.yMax - extent.yMin;
-    }
-    entry[str] = extent;
-    return extent;
-  }
-  /**
-   * Retrieve the total width of `text` in `em` units.
-   */
-  getWidthForTextInEm(text: string): number {
```

Comment:
Is your measureText() related to this?

Reply:
Is this question for me?

Reply:
Yes, I was just thinking about whether these methods have any uses. Your new v5 approach relies heavily on a measureText() method. I wonder if others also have use for methods that are similar (for layout / UI purposes).

This method is called getWidthForTextInEm() so it sounds like it might be a useful utility method.

Reply:
In any case, if you think this class has no uses for now, we can delete it. But if anyone complains, we can always restore it (from the most recent version, or from the v4 repository).

Reply:
In VexFlow v4 Glyph focuses on SMuFL metrics and TextFormatter on other Fonts (i.e.: Roboto Slab') In VexFlow v5 we considered both glyphs and standard characters text and it is all handled by Element.